Use proper cmake build options

G10h4ck opened the pull request:
Require linker dependencies only if executables are built
Fix missing library linking regression introduced in my previous PR

Sorry to not have included this in latest PR, you where much faster then I expected!

truvorskameikin commented on CMakeLists.txt:
```diff
@@ -1,5 +1,8 @@
 cmake_minimum_required(VERSION 2.8)
 
+option( BUILD_EXAMPLE "Build example application."                  ON )
```
Let's be consistent with coding style:
- no spaces before and after parentheses
- no long horizontal spacing

truvorskameikin commented on CMakeLists.txt:
```diff
@@ -11,29 +14,38 @@ set(LIB_HEADERS
-		discovery_tool.cpp)
-	set(DISCOVERY_TOOL_LIBS
-		udp-discovery)
+if ( BUILD_EXAMPLE OR BUILD_TOOL )
```
Same as above.

truvorskameikin commented on CMakeLists.txt:
```diff
@@ -11,29 +14,38 @@ set(LIB_HEADERS
 	target_link_libraries(udp-discovery-example ${DISCOVERY_EXAMPLE_LIBS})
+endif()
+
+if ( BUILD_TOOL )
```
Same as above.

G10h4ck commented:
For me it is acceptable to turn them OFF by default, but in my personal experience it have been very useful to have them complied by default with just `cmake . ; make` without wondering about options to test it actually worked ;) But this could be easily overcome updating the README too.

I am implementing broadcast discovery into RetroShare https://retroshare.cc and was looking for a cross-platform lightweight solution without reinventing the wheel (handling the cumbersome socket API defferences on all platforms) and without the bloat and unreliability of bonjour/avahi/zeroconf/mdns, during this quest I have found your project that seems exactly what I was looking for, at first I was a bit scared by last commit being reported in 2017, but then saw some more activity in the develop branch and I have been positively surprised by your responsiveness to pull requests :)

truvorskameikin commented:
That is great that you find this library useful! Let's finish with formatting and merge this request without changing build options.

Please feel free to ask for adding reasonable features and to offer help here.

G10h4ck commented:
Too late, I have already both fixed formatting, changed default build options, and updated documentation accordingly ;)

I am glad to contribute to this library :)

truvorskameikin commented:
Thank you!
